### PR TITLE
Print dangling comments in blocks

### PR DESCRIPTION
**Pull request by jlongster**
I'm still thinking through the best way to handle comments. For now, I'm going to fix a few smaller bugs first so I can get a better understanding of all the edge cases. This one fixes the empty block problem.

Once I see look at more use cases and how the AST is structured I'll see if there's a better way to handle comments generally.

### CODE REVIEW

File: tests/more_path/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -102,7 +102,8 @@ function goofy() {
   var x = g();
   if (typeof x == \"function\") {
     x();
-  } else {
+  } else { // if (typeof x == \'number\') {
```
Comment (jlongster) [inline tests/more_path/__snapshots__/jsfmt.spec.js.snap]: This is actually correct, the original source had it on the same line

File: src/printer.js [hunk 2]
```diff
@@ -510,6 +510,12 @@ function genericPrintNoParens(path, options, print) {
       if (hasContent) {
         parts.push(indent(options.tabWidth, concat([ hardline, naked ])));
       }
+      else if(n.comments) {
+        parts.push(
+          indent(options.tabWidth,
```
Comment (vjeux) [inline src/printer.js]: the awesome part about prettier is that I don't need to tell you

nit: should be

```js
indent(
  options.tabWidth,
  comments.printDanglingComments(path, print, options))
)
```

**Comment by vjeux** [comment]
Looks good to me!

File: src/printer.js (hunk 2 above)
Comment (jlongster) [inline src/printer.js]: yeah, because we haven't automatically hooked it up I just code whatever and know it'll be formatted at some point. I'll change this though!

We're probably stable enough to try to enforce PRs to be formatted. I should at least turn on Emacs integration again. We weren't stable enough before but we probably are now.
